spec/controllers/messages_controller_spec.rb

```diff
@@ -60,6 +60,34 @@
       expect(assigns(:message).recipient_id).to eq(recipient.id)
     end
 
+    it "handles provided blocked user" do
+      block = create(:block)
+      login_as(block.blocking_user)
+      get :new, params: { recipient_id: block.blocked_user_id }
+      expect(response.status).to eq(200)
+      expect(assigns(:message).recipient_id).to be_nil
+    end
+
+    it "handles provided blocking user" do
+      block = create(:block)
+      login_as(block.blocked_user)
+      get :new, params: { recipient_id: block.blocking_user_id }
+      expect(response.status).to eq(200)
+      expect(assigns(:message).recipient_id).to be(block.blocking_user_id)
+    end
+
+    it "hides blocked users" do
+      user = create(:user)
+      login_as(user)
+      blocking = create_list(:block, 2, blocked_user: user).pluck(:blocking_user_id)
```

Comment:
You could simply map this to `:blocking_user`, and then you don't need to do `User.find(blocking)` below.